### PR TITLE
Allow customize DisconnectTimeout for ASRS timeout idle connections

### PR DESCRIPTION
**Pull request by JialinXin**
support #846

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs [hunk 1]
```diff
@@ -209,7 +209,7 @@ private IEnumerable<Claim> BuildClaims(IOwinContext owinContext, IRequest reques
             var user = owinContext.Authentication?.User;
             var userId = _provider?.GetUserId(request);
 
-            var claims = ClaimsUtility.BuildJwtClaims(user, userId, GetClaimsProvider(owinContext), _serverName, _mode, _enableDetailedErrors, _endpointsCount);
+            var claims = ClaimsUtility.BuildJwtClaims(user, userId, GetClaimsProvider(owinContext), _serverName, _mode, _enableDetailedErrors, _endpointsCount, _disconnectTimeout);
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs:212]: should we use builder pattern?
I find wangpeng also add an parameter in this build.

File: src/Microsoft.Azure.SignalR/ServiceOptionsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace Microsoft.Azure.SignalR
+{
+    internal static class ServiceOptionsExtensions
+    {
+        public static void Validate(this ServiceOptions options)
+        {
+            if (options.DisconnectTimeoutInSeconds.HasValue &&
+                (options.DisconnectTimeoutInSeconds < 1 || options.DisconnectTimeoutInSeconds > 300))
+            {
+                throw new ArgumentOutOfRangeException("DisconnectTimeoutInSeconds", "Value should be [1,300].");
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR/ServiceOptionsExtensions.cs:15]: * ArgumentException, the argument is 'options'
* Can we check it when set the property `DisconnectTimeoutInSeconds`?

Comment (JialinXin) [reply src/Microsoft.Azure.SignalR/ServiceOptionsExtensions.cs:15]: Not able to move this into ServiceOptionsSetup cause the value is not coming from file configuration but code option settings.

AspNetCore support [`IOptionsValidate`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1) after 2.2 which is not compatible with 2.1, so choose to use `PostConfigure` to do the validation.

Comment (JialinXin) [reply src/Microsoft.Azure.SignalR/ServiceOptionsExtensions.cs:15]: I think `ArgumentOutOfRangeException` should be more appropriate for our scenario. https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception?view=netcore-3.1#remarks

File: src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs (hunk 1 above)
Comment (JialinXin) [reply src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs:212]: Good idea. Will use another PR to do the refactor.

**Pull request merged**